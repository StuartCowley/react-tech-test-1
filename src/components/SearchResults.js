import React from "react";
import PropTypes from "prop-types";

import "../styles/SearchResults.css";

const SearchResults = ({ results, userHasSearched }) => {
  if (results.length === 0 && userHasSearched === false) {
    return null;
  }

  if (results.length === 0 && userHasSearched === true) {
    return <p>No results found.</p>;
  } else {
    return (
      <div className="search-results">
        {results.map((image) => (
          <img
            className="data-img"
            src={image}
            alt="space-img"
            data-testid="data-img"
          />
        ))}
      </div>
    );
  }
};

SearchResults.propTypes = {
  results: PropTypes.array,
};

export default SearchResults;
