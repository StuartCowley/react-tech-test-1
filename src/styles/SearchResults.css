.search-results,
p {
  text-align: center;
  width: 80%;
  margin: 5% 10% 0 10%;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.data-img {
  width: 300px;
  height: 250px;
  flex: center;
  margin: 20px;
  border-radius: 12px;
}

.data-img:hover {
  transform: scale(1.2);
  cursor: pointer;
}
