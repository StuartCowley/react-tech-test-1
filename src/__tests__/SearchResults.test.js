import React from "react";
import SearchResults from "../components/SearchResults";
import { render, screen } from "@testing-library/react";

describe("SearchResults", () => {
  const validProps = {
    results: [
      {
        data: [
          {
            title: "ROBOTS TO ROCKET CITY",
            nasa_id: "MSFC-1600354",
          },
        ],
        links: [
          {
            href: "https://images-assets.nasa.gov/image/MSFC-1600354/MSFC-1600354~thumb.jpg",
          },
        ],
      },
      {
        data: [
          {
            title: "ROBOTS TO ROCKET CITY",
            nasa_id: "MSFC-1600353",
          },
        ],
        links: [
          {
            href: "https://images-assets.nasa.gov/image/MSFC-1600353/MSFC-1600353~thumb.jpg",
          },
        ],
      },
    ],
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchResults results={validProps.results} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("returns the search results", () => {
    render(
      <SearchResults
        results={validProps.results}
        userHasSearched={validProps.userHasSearched}
      />
    );
    const imageResults = screen.getAllByAltText("space-img");

    for (let x = 0; x < imageResults.length; x += 1) {
      expect(imageResults[x]).toHaveClass("data-img");
    }

    expect(imageResults).toHaveLength(2);
  });
});
