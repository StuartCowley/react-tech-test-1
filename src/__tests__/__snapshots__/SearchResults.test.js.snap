// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <img
      alt="space-img"
      class="data-img"
      data-testid="data-img"
      src="[object Object]"
    />
    <img
      alt="space-img"
      class="data-img"
      data-testid="data-img"
      src="[object Object]"
    />
  </div>
</DocumentFragment>
`;
